File: EASY-DOC/legal-ai/server/features/risk/risk.model.js
import mongoose from 'mongoose';

const riskSchema = new mongoose.Schema({
  documentId: { type: String, required: true, unique: true },
  riskScore: { type: Number, required: true, min: 0, max: 100 },
  flags: [{
    type: { type: String },
    term: { type: String },
    severity: { type: String, enum: ['low', 'medium', 'high'] },
    context: { type: String },
  }],
  summary: { type: String },
  analyzedAt: { type: Date, default: Date.now },
});

export const Risk = mongoose.model('Risk', riskSchema);
export default Risk;

File: EASY-DOC/legal-ai/nlp-service/app/features/risk/risk_model.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime


class RiskFlag(BaseModel):
    type: str
    term: str
    severity: str = Field(..., pattern="^(low|medium|high)$")
    context: Optional[str] = None


class RiskRequest(BaseModel):
    document_id: str


class RiskResponse(BaseModel):
    risk_score: int = Field(..., ge=0, le=100)
    flags: List[RiskFlag]
    summary: str
    analyzed_at: datetime = Field(default_factory=datetime.utcnow)


class RiskAnalysisResult(BaseModel):
    document_id: str
    score: int
    flags: List[dict]
    llm_summary: str
    processing_time_ms: Optional[float] = None
